R/joyn-merge.R

```diff
@@ -429,21 +439,49 @@ joyn <- function(x,
                          .xreport = NULL,
                          .yreport = NULL)
 
-
   if (sort) {
     setorderv(x, by, na.last = na.last)
     setattr(x, 'sorted', by)
   }
 
   ## Rename by variables -----
 
-  if (!is.null(fixby$xby)) {
-    data.table::setnames(x, fixby$tempkey, fixby$xby)
-    by <- fixby$xby
-    # not necessary
-    # setnames(y, fixby$tempkey, fixby$yby)
+  # in output
+  if (any(grepl(pattern = "keyby", x = names(x)))) {
+    data.table::setnames(x,
+                         old = names(x)[grepl(pattern = "keyby",
+                                              x       = names(x))],
+                         new = xbynames)
+  }
+
+
+  # Change names back for inputs------------------------------
+  if (any(grepl(pattern = "keyby", x = names(x_original)))) {
+
+    knames <- names(x_original)[grepl(pattern = "keyby",
+                                      x       = names(x_original))]
+    knames <- knames[order(knames)]
+
+    data.table::setnames(x_original,
+                         old = knames,
+                         new = xbynames)
   }
 
+  if (any(grepl(pattern = "keyby", x = names(y_original)))) {
+
+    knames <- names(y_original)[grepl(pattern = "keyby",
+                                      x       = names(y_original))]
+    knames <- knames[order(knames)]
+
+    data.table::setnames(y_original,
+                         old = knames,
+                         new = ybynames)
+
+    if (all(names(y_original) %in% ynames)) {
+      colorderv(y_original,
+                neworder = ynames)
+    }
+  }
 
```

Comment:
When match_type is `m:m` this never hold because of the `as.data.table()` above, right? Is this by intention?